Removed F1 targets from the target lists used for CI, added warning about deprecation.

mikeller opened the pull request.

AndersHoglund commented:
`make all ` is used by Jenkins and will still make legacy unsupported targets, with warnings. We need either to filter out `$(LEGACY_TARGET)` from `$(VALID_TARGETS)` so goal `all` really means `all_supported`.
Or add a goal `all_supported` that Jenkins (and others) can use. And an `all_target_groups` goal. Same as `all_supported`.
See diff below.

```
$ git diff
diff --git a/Makefile b/Makefile
index 96b02ec..b2b98f1 100644
--- a/Makefile
+++ b/Makefile
@@ -318,6 +318,12 @@ $(OBJECT_DIR)/$(TARGET)/%.o: %.S
 ## all               : Build all valid targets
 all: $(VALID_TARGETS)

+## all_supported     : Build all supported targets
+all_supported: $(SUPPORTED_TARGETS)
+
+## all_target_groups : Build all supported target groups
+all_target_groups: $(ALL_TARGET_GROUPS)
+
 ## official          : Build all official (travis) targets
 official: $(OFFICIAL_TARGETS)

@@ -446,6 +452,7 @@ help: Makefile make/tools.mk
 ## targets           : print a list of all valid target platforms (for consumption by scripts)
 targets:
        $(V0) @echo "Valid targets:      $(VALID_TARGETS)"
+       $(V0) @echo "Supported targets:  $(SUPPORTED_TARGETS)"
        $(V0) @echo "Target:             $(TARGET)"
        $(V0) @echo "Base target:        $(BASE_TARGET)"
        $(V0) @echo "targets-group-1:    $(GROUP_1_TARGETS)"
diff --git a/make/targets.mk b/make/targets.mk
index 772e150..af6f380 100644
--- a/make/targets.mk
+++ b/make/targets.mk
@@ -109,6 +109,8 @@ LEGACY_TARGETS := \
        NAZE

 GROUP_OTHER_TARGETS := $(filter-out $(GROUP_1_TARGETS) $(GROUP_2_TARGETS) $(GROUP_3_TARGETS) $(GROUP_4_TARGETS) $(LEGACY_TARGETS), $(VALID_TARGETS))
+ALL_TARGET_GROUPS := $(GROUP_1_TARGETS) $(GROUP_2_TARGETS) $(GROUP_3_TARGETS) $(GROUP_4_TARGETS) $(GROUP_OTHER_TARGETS))
+SUPPORTED_TARGETS := $(ALL_TARGET_GROUPS)

 ifeq ($(filter $(TARGET),$(ALT_TARGETS)), $(TARGET))
 BASE_TARGET    := $(firstword $(subst /,, $(subst ./src/main/target/,, $(dir $(wildcard $(ROOT)/src/main/target/*/$(TARGET).mk)))))

```

AndersHoglund commented:
First option ( `all` meaning "all_supported") would maybe also require adding a `legacy` and a `legacy_clean` goal. 
I think I would like this option better, `make all` would then still work as before, only with less targets.

AndersHoglund commented:
Tried filtering out `$(LEGACY_TARGETS)` from `$(VALID_TARGETS)` and that makes it impossible to build any of the F1 targets. Might as well remove all F1s from the `src/main/target` dir

Also, default build target needs to change, still NAZE. 

```
$ git diff
diff --git a/Makefile b/Makefile
index 96b02ec..1375209 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@
 #

 # The target to build, see VALID_TARGETS below
-TARGET    ?= NAZE
+TARGET    ?= BETAFLIGHTF3

```

mikeller commented:
Good point, @AndersHoglund. I've updated the pull request so that `all` now builds only the supported targets, and added `unsupported` and `all_with_unsupported` as new make targets. (Also, BETAFLIGHTF3 is now the default make target.)

AndersHoglund commented:
Thnx. Great.
But really, why not simply remove all F1s? From Makefiles and file system. It is all there in the git repo and available on old tags and maint branches, if someone wants them. 
We should not to add complexity and support for the un-supported. Enough spaghetti ans target lists as it is.  Just remove them.

MJ666 commented:
I would definitely keep to code in to allow skilled users to continue to make custom builds for the F1 targets as long as it will not hurt future development. There is may be a time in the future to remove the F1 code but I don't think this is now already. Many of the F1 targets have still space if advanced features get removed. For many brushed and acro flyers F1 FC's still perform well.

mikeller commented:
This PR is only the very beginning of the removal of support for F1, intended to avoid PRs failing to build if they break something in an F1 target.

I don't see the removal of the F1 target definitions as something that has any urgency - they are isolated, and their removal will not do anything to reduce the complexity of the code that is currently supported.

I don't think that removal of code that is conditional on `STM32F1` should be done in an 'all at once', find / replace type fashion - often the removal of this code will allow for simplification of the code that goes beyond simply deleting the lines that are conditional, and these simplifications are best done at the time the conditional code is removed, or finding the places with potential for simplification will be hard. So this is best left to be done in a series of individual PRs.
